readme.md

```diff
@@ -47,126 +62,497 @@ This package exports [ESM](https://gist.github.com/sindresorhus/a39789f98801d908
 
 ## Usage
 
-This is a basic example using OpenAI's [text-embedding-ada-002](https://platform.openai.com/docs/guides/embeddings) embedding model and a [Pinecone](https://www.pinecone.io/) datastore to index and query a set of documents.
+Here's a basic example of how to use the ChatModel:
 
-```ts
-import 'dotenv/config';
-import { EmbeddingModel } from '@dexaai/dexter/model';
-import { PineconeDatastore } from '@dexaai/dexter/datastore/pinecone';
+```typescript
+import { ChatModel } from '@dexaai/dexter';
 
-async function example() {
-  const embeddingModel = new EmbeddingModel({
-    params: { model: 'text-embedding-ada-002' },
-  });
+const chatModel = new ChatModel({
+  params: { model: 'gpt-3.5-turbo' },
+});
 
-  const store = new PineconeDatastore({
-    contentKey: 'content',
-    embeddingModel,
+async function main() {
+  const response = await chatModel.run({
+    messages: [{ role: 'user', content: 'Tell me a short joke' }],
   });
+  console.log(response.message.content);
+}
+
+main().catch(console.error);
+```
+
+## Examples
+
+### Chat Completion with Streaming
+
+```typescript
+import { ChatModel, MsgUtil } from '@dexaai/dexter';
+
+const chatModel = new ChatModel({
+  params: { model: 'gpt-4' },
+});
 
-  await store.upsert([
-    { id: '1', metadata: { content: 'cat' } },
-    { id: '2', metadata: { content: 'dog' } },
-    { id: '3', metadata: { content: 'whale' } },
-    { id: '4', metadata: { content: 'shark' } },
-    { id: '5', metadata: { content: 'computer' } },
-    { id: '6', metadata: { content: 'laptop' } },
-    { id: '7', metadata: { content: 'phone' } },
-    { id: '8', metadata: { content: 'tablet' } },
-  ]);
-
-  const result = await store.query({ query: 'dolphin' });
-  console.log(result);
+async function main() {
+  const response = await chatModel.run({
+    messages: [MsgUtil.user('Write a short story about a robot learning to love')],
+    handleUpdate: (chunk) => {
+      process.stdout.write(chunk);
+    },
+  });
+  console.log('\n\nFull response:', response.message.content);
 }
+
+main().catch(console.error);
 ```
 
-## Docs
+### Extracting Structured Data
+
+```typescript
+import { ChatModel } from '@dexaai/dexter';
+import { createExtractFunction } from '@dexaai/dexter/extract';
```

Comment:
from a DX standpoint, this second import line for this type of lib is totally unnecessary.

I know you're going to disagree w/ my comment above, but it really makes the DX for first-time users much nicer to only have one required import. They're not letting their editor figure out imports when they don't even know what the imports are yet.